lib/sherlock_homes/trulia.rb

```diff
@@ -64,27 +64,62 @@ def parse
 
     class SummaryParser < AbstractParser
       def parse
-        result = []
+        result = Hash.new
         @data.css('li').each do |li|
-          result << li.text.strip
+          text = li.text.strip
+          next if text.eql? "Edit Home Facts"
+          result.merge! BulletItemParser.parse(text)
         end
-        result.delete("Edit Home Facts")
         result
       end
     end
 
     class PublicRecordsParser < AbstractParser
       def parse
-        result = []
+        result = Hash.new
         @data.css('li').each do |group|
           group.css('ul > li').each do |li|
-            result << li.text.strip
+            result.merge! BulletItemParser.parse(li.text.strip)
           end
         end
         result
       end
     end
 
```

Comment:
What is AbstractParser buying us here?  Seems like:
```ruby
module BulletItemParser
  def self.parse(data)
    # do things
  end
end

would be sufficient. Or if you wanted to use lambdas:

```ruby
    ImageUrlParser = lambda do |data|
      element = data.first
      style = element.attributes['style'].value
      /background-image:url\('(.*)'\);/.match(style)[1]
    end
  ImageUrlParser.call(data)
```